File: microservices/old-finder/models/supplierBox.py
from database.db import db
from flask_mongoengine import BaseQuerySet
from slugify import slugify, Slugify, UniqueSlugify
from models.box import Box
import uuid
import datetime

def id(): return str(uuid.uuid4())

class SupplierBox(db.Document):
    iso = db.StringField(required=True)
    row_id = db.StringField(required=False)
    sort = db.IntField(required=False, default=0)
    tenant_id = db.StringField(required=True)
    tenant_name = db.StringField(required=False)
    sku = db.StringField(required=False, default='')
    name = db.StringField(required=True, default='', unique_with=('tenant_id'))
    display_name = db.ListField(required=True, default=[])
    system_key = db.StringField(required=True)
    input_type = db.StringField(required=False, default='')
    incremental = db.BooleanField(required=True, default=False)
    select_limit = db.IntField(required=True, default=0)
    option_limit = db.IntField(required=True, default=0)
    sqm = db.BooleanField(required=True, default=False)
    slug = db.StringField(required=True, default='', unique_with=('tenant_id'))
    linked = db.ReferenceField(Box)
    description = db.StringField(required=False, default='')
    media = db.ListField()
    shareable = db.BooleanField(default=True)
    published = db.BooleanField(default=True)
    created_at = db.DateTimeField(default=datetime.datetime.now, required=True)
    additional = db.DictField(required=False, default={})

    meta = {
        'indexes': ['-created_at', 'slug', 'name'],
        'ordering': ['-created_at'],
        'collection': 'supplier_boxes',
        'queryset_class': BaseQuerySet
    }

    def __setattr__(self, key, value):
        super(SupplierBox, self).__setattr__(key, value)
        if key == 'name':
            self.slug = slugify(self.name, to_lower=True)

File: microservices/price/price/Models/SupplierBox.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;


const SupplierBoxSchema = Schema({
    iso:{type: String, required: true},
    row_id:{type: String, required: false},
    sort:{type: Number, required: false, default: 0},
    tenant_id:{type: String, required: true},
    tenant_name:{type: String, required: false},
    sku:{type: String, required: false, default:''},
    name:{type: String }, // = db.String(required=True, default='', unique_with=('tenant_id'))
    display_name:{type: [], required: true}, // = db.[](required=True, default=[])
    system_key:{type: String, required: true}, // = db.String(required=True)
    input_type:{type: String, required: false}, // = db.String(required=False, default='')
    incremental:{type: Boolean, required: true}, // = db.Boolean(required=True, default=False)
    select_limit:{type: Number, required: true}, // = db.Number(required=True, default=0)
    option_limit:{type: Number, required: true}, // = db.Number(required=True, default=0)
    sqm:{type: Boolean, required: true}, // = db.Boolean(required=True, default=False)
    slug:{type: String, required: true}, // = db.String(required=True, default='', unique_with=('tenant_id'))
    appendage:{type: String, required: true}, // = db.String(required=True, default='', unique_with=('tenant_id'))
    calculation_type:{type: String, required: true}, // = db.String(required=True, default='', unique_with=('tenant_id'))
    // linked:{type: ReferenceField}, // = db.ReferenceField(Box)
    description:{type: String, required: false}, // = db.String(required=False, default='')
    media:{type: []}, // = db.[]()
    shareable:{type: Boolean, default: true}, // = db.Boolean(default=True)
    published:{type: Boolean, default: true}, // = db.Boolean(default=True)
    created_at:{type: Date, required: true}, // = db.DateTimeField(default=datetime.datetime.now, required=True)
    // additional:{type: DictField, required: true}, // = db.DictField(required=False, default={})
    additional:{type: String, required: true},
});


module.exports = SupplierBox = mongoose.model("supplier_boxes", SupplierBoxSchema);
